### ISSUE TITLE
':branch' should default to a String type

### ISSUE THREAD

**Issue by griley** [capistrano/capistrano#1499]
Currently, when no `:branch` is specified as part of a deploy task Capistrano sets a default here:

https://github.com/capistrano/capistrano/blob/master/lib/capistrano/defaults.rb#L11

... which is of type `Symbol`.  This is contrary to the [docs](http://capistranorb.com/documentation/getting-started/configuration/) and is a different type than what would be used if the the `:branch` was set via `ENV` variables:

```
set :branch, ENV["REVISION"] || ENV["BRANCH_NAME"] || "master"
```

Because `fetch(:branch)` could return a `Symbol` or a `String`, every capistrano task is required to use `to_s` when doing comparisons.  Obviously, I can do this in my own cap tasks, but I've seen this cause issue in capistrano deployment add-ons.

**Comment by leehambley** [comment]
How about a PR @griley I'd be more than happy to merge it.

**Comment by Kriechi** [comment]
@griley please close this issues - as it should already be resolved by #1504.
